DOL.WHD.Section14c.Web/src/modules/app.js

```diff
@@ -37,24 +41,49 @@ let app = angular.module('14c', [
   ngResource,
   ngRoute,
   ngSanitize,
+  toastr,
   'angularMoment',
   'ngMask',
   'ngCookies',
   'dataGrid',
   'pagination'
 ]);
 
+// augment angular exception handler
+app.provider('$exceptionHandler', { $get: function( errorLogService ) {
+  return( errorLogService );
+}});
+
 app
   .directive('dolFooter', downgradeComponent({ component: DolFooterComponent }))
   .directive('dolHeader', downgradeComponent({ component: DolHeaderComponent }))
-  .directive(
-    'helloWorld',
-    downgradeComponent({ component: HelloWorldComponent })
-  )
-  .directive(
-    'uiLibrary',
-    downgradeComponent({ component: UiLibraryComponent })
-  );
+  .directive('helloWorld', downgradeComponent({ component: HelloWorldComponent }))
+  .directive('uiLibrary', downgradeComponent({ component: UiLibraryComponent }))
+  .factory('loggingService', downgradeInjectable(LoggingService))
+  .factory("errorLogService", function( $log, loggingService ) {
+      // log the given error to the remote server.
+      function log( exception, cause ) {
+          // Pass off the error to the default error handler
+          // on the AngualrJS logger. This will output the
+          // error to the console (and let the application
+          // keep running normally for the user).
+          $log.error.apply( $log, arguments );
+          // prevents the same client from
+          // logging the same error over and over again
+          try {
+              var errorMessage = new customError(exception.toString(), "Error")
+              // Log the JavaScript error to the server.
+
+              loggingService.addLog(errorMessage)
+          } catch ( loggingError ) {
+              // For Developers - log the log-failure.
+              $log.warn( "Error logging failed" );
+              $log.log( loggingError );
+          }
+      }
+      // Return the logging function.
+      return( log );
+  });
```

Comment:
I'm wondering if this `factory` function can be a static on the `LoggingService` class, or another export from that file.  I just worry about putting too much stuff in `app.js`.  It already has a lot in it.